server/cluster/cluster.go

```diff
@@ -246,27 +272,57 @@ func (c *Cluster) CreateSchema(ctx context.Context, schemaName string, schemaID
 	return schema, nil
 }
 
-func (c *Cluster) CreateTable(ctx context.Context, schemaName string, shardID uint32,
-	tableName string, tableID uint64,
-) (*Table, error) {
+func (c *Cluster) CreateTable(ctx context.Context, nodeName string, schemaName string, tableName string) (*Table, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
 	// Check provided schema if exists.
-	schema, exists := c.GetSchema(schemaName)
+	schema, exists := c.getSchemaLocked(schemaName)
 	if !exists {
 		return nil, ErrSchemaNotFound.WithCausef("schemaName", schemaName)
 	}
 
-	// Save Table in storage.
+	// check if exists
+	table, exists := c.getTableLocked(schemaName, tableName)
+	if exists {
+		return table, nil
+	}
+
+	// create new schemasCache
+	shardID, err := c.pickOneShardOnNode(nodeName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "clusters AllocTableID, "+
+			"clusterName:%s, schemaName:%s, tableName:%s, nodeName:%s", c.Name(), schemaName, tableName, nodeName)
+	}
+
+	// alloc table id
+	tableID, err := c.allocTableID(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cluster AllocTableID, "+
+			"schemaName:%s, tableName:%s", schemaName, tableName)
+	}
+
+	// Save table in storage.
 	tablePb := &clusterpb.Table{Id: tableID, Name: tableName, SchemaId: schema.GetID(), ShardId: shardID}
-	err := c.storage.CreateTable(ctx, c.clusterID, schema.GetID(), tablePb)
+	tablePb, err = c.storage.CreateTable(ctx, c.clusterID, schema.GetID(), tablePb)
 	if err != nil {
 		return nil, errors.Wrap(err, "clusters CreateTable")
 	}
 
+	// Update shardTopology in storage.
+	shardTopologies, err := c.storage.ListShardTopologies(ctx, c.clusterID, []uint32{shardID})
+	if len(shardTopologies) != 1 {
+		return nil, errors.Wrapf(err, "clusters CreateTable, shard has more than one shardTopology, shardID:%d, shardTopologies:%v",
```

Comment:
The `err` is actually nil, and maybe we should define a leaf error to tell the cause.